src/state/operator/selectors.js

```diff
@@ -58,16 +102,22 @@ const isMemberOfGroups = ( userID, groups ) => compose(
 	mergeAll
 )( groups )
 
+/**
+ * Returns a list of available operators for the given locale and groups,
+ * sorted by priority where the first operator in the list should be
+ * assigned the next chat.
+ */
 export const getAvailableOperators = ( locale, groups, state ) => compose(
 	flatten,
 	operators => map( group => compose(
-		sort( compare ),
+		sort( compareOperatorPriority ),
 		map( user => merge( user, {
 			percentAvailable: percentAvailable( user ),
 			totalAvailable: totalAvailable( user )
 		} ) ),
 		filter( ( { status, online, load, capacity, active, id } ) => {
-			if ( !online || status !== STATUS_AVAILABLE ) {
+			const isAvailable = status === STATUS_AVAILABLE || status === STATUS_RESERVE;
```

Comment:
Maybe `isAvailable` should be a function thats shared by `getAvailableOperators` and `compareOperatorPriority`